src/utils/helpers/application/buildQueryFilters.ts

```diff
@@ -0,0 +1,119 @@
+import { DateTime } from 'luxon'
+import { QueryObject } from '../../hooks/useRouter'
+import { ApplicationStatus, ApplicationOutcome } from '../../../utils/generated/graphql'
+
+export default function buildQueryFilters(filters: QueryObject) {
+  const graphQLfilter = Object.entries(filters).reduce((filterObj, [key, value]) => {
+    if (!mapQueryToFilterField?.[key]) return filterObj
+    return { ...filterObj, ...mapQueryToFilterField[key](value) }
+  }, {})
+  return graphQLfilter
```

Comment:
Yeah, I kind of thought that since the expression is so long and hard to understand that having a variable name describing exactly what it results in might be helpful. Not really fussed either way.